content-cache/charmcraft.yaml

```diff
@@ -13,44 +13,37 @@ platforms:
-      # Get upstreal module (the one we are interested in)
-      git clone https://github.com/openresty/lua-upstream-nginx-module.git --depth 1
-      # Build NGINX with the lua-upstream-nginx-module
-      ./configure --add-dynamic-module=lua-upstream-nginx-module --prefix=/usr/local/nginx --with-cc-opt="-I/usr/include/luajit-2.1 -I$(pwd)/lua-nginx-module/src/api/ -g -O2 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -ffile-prefix-map=/build/nginx-DlMnQR/nginx-1.24.0=. -flto=auto -ffat-lto-objects -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -fcf-protection -fdebug-prefix-map=/build/nginx-DlMnQR/nginx-1.24.0=/usr/src/nginx-1.24.0-2ubuntu7.1 -fPIC -Wdate-time -D_FORTIFY_SOURCE=3" --with-ld-opt='-Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -Wl,-z,relro -Wl,-z,now -fPIC' --prefix=/usr/share/nginx --conf-path=/etc/nginx/nginx.conf --http-log-path=/var/log/nginx/access.log --error-log-path=stderr --lock-path=/var/lock/nginx.lock --pid-path=/run/nginx.pid --modules-path=/usr/lib/nginx/modules --http-client-body-temp-path=/var/lib/nginx/body --http-fastcgi-temp-path=/var/lib/nginx/fastcgi --http-proxy-temp-path=/var/lib/nginx/proxy --http-scgi-temp-path=/var/lib/nginx/scgi --http-uwsgi-temp-path=/var/lib/nginx/uwsgi --with-compat --with-debug --with-pcre-jit --with-http_ssl_module --with-http_stub_status_module --with-http_realip_module --with-http_auth_request_module --with-http_v2_module --with-http_dav_module --with-http_slice_module --with-threads --with-http_addition_module --with-http_flv_module --with-http_gunzip_module --with-http_gzip_static_module --with-http_mp4_module --with-http_random_index_module --with-http_secure_link_module --with-http_sub_module --with-mail_ssl_module --with-stream_ssl_module --with-stream_ssl_preread_module --with-stream_realip_module --with-http_geoip_module=dynamic --with-http_image_filter_module=dynamic --with-http_perl_module=dynamic --with-http_xslt_module=dynamic --with-mail=dynamic --with-stream=dynamic --with-stream_geoip_module=dynamic
-      make
-      cp objs/ngx_http_lua_upstream_module.so /root/stage
+
+      # Fetch the module to compile
+      repo=lua-upstream-nginx-module
+      rm -Rf $repo
+      git clone https://github.com/openresty/$repo
+      cd $repo
+      git checkout 07b11be7a2e6e1b75209745915a84d281737d4d7
+
+      # Build package
+      cp -R ../debian .
+      curl https://raw.githubusercontent.com/openresty/lua-nginx-module/f44c18859050be5af314bae8f05660478af5e31d/src/api/ngx_http_lua_api.h -O --output-dir debian/
+      dpkg-buildpackage -us -uc
+      cp ../libnginx-mod-http-upstream_0.1_amd64.deb /root/stage
     prime:
-      - ngx_http_lua_upstream_module.so
+      - libnginx-mod-http-upstream_0.1_amd64.deb
   nginx-upstream-healthcheck:
     plugin: dump
     source: https://github.com/openresty/lua-resty-upstream-healthcheck.git
+      # source-commit: 7924b2abbcc3269423611e6d7aa66be28f4076f0
```

Comment:
Should this be commented out?

Reply:
No indeed. This is the current version so it doesn't change the content of the charm, but I will re-run the build & tests to ensure I did not comment it for some issue with charmcraft.

Reply:
@jdkandersson tests are ok if you can have a look again ;)